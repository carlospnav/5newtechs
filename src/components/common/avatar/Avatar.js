import styled from 'styled-components';
import { avatarProps } from 'theme/utils';


/* Avatar Component - Renders a small or large
  avatar image based on the small prop passed to it.
*/
const Img = styled.img`
  object-fit: contain;
`

export const BigImg = styled(Img).attrs({
  src: "/static/avatar/icn-avatar-big.svg",
  width: "300px",
  height: "283px"
})`
`

export const SmallImg = styled(Img).attrs({
  src: "/static/avatar/icn-avatar-small.svg",
  width: "140px",
  height: "132.6px"
})`
`

export default function({ small = false }) {
  return (
    small ? <SmallImg /> : <BigImg />
  )
}
