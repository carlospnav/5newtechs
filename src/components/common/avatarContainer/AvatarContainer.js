import styled from "styled-components"
import { flexGroup } from "theme/utils"
import Avatar from "../avatar/Avatar"

const Container = styled.div`
  ${flexGroup("column", null, "center")};
  padding: 40px;
`
const Name = styled.h3`
  color: #a0a9ab;
  font-size: 18px;
  margin: 20px 0 0 0;
`

export default function({item}) {
  return (
    <Container>
      <Avatar small/>
      <Name>{item.name}</Name>
    </Container>
  )
}
