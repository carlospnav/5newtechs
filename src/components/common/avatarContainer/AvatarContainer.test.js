import React from "react"
import { shallow } from "enzyme"
import AvatarContainer, { Name } from "./AvatarContainer"
import Avatar from "../avatar/Avatar"
import "jest-styled-components"

function avatarContainer(){
  return shallow(<AvatarContainer name="foo"/>)
}

describe("AvatarContainer", () =>{
  let wrapper;

  beforeEach(() => wrapper = avatarContainer())
  afterEach(() => wrapper = undefined )

  it("renders an AvatarContainer properly", () => {
    expect(wrapper.dive()).toMatchSnapshot()
  })

  it("renders the name component properly.", () => {
    expect(wrapper.find(Name).dive()).toMatchSnapshot()
    expect(wrapper.find(Name).dive().text()).toEqual("foo")
  })

  it("always renders an `Avatar`", () => {
     expect(wrapper.find(Avatar)).toHaveLength(1);
  });
})