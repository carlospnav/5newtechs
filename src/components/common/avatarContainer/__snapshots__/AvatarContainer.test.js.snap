// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarContainer renders an AvatarContainer properly 1`] = `
.c0 {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  padding: 20px;
}

@media (min-width:1024px) {
  .c0 {
    width: 45%;
    padding: 40px 10px;
  }
}

<section
  className="c0"
>
  <Component />
  <styled.h2>
    foo
  </styled.h2>
</section>
`;

exports[`AvatarContainer renders the name component properly. 1`] = `
.c0 {
  color: #a0a9ab;
  font-size: 26px;
  font-weight: 300;
  color: white;
  font-family: 'Roboto',sans-serif;
  margin: 60px 0 30px 0;
}

@media (min-width:1024px) {
  .c0 {
    margin: 30px 0;
  }
}

<h2
  className="c0"
>
  foo
</h2>
`;
